lib/saml_idp/request.rb

```diff
@@ -71,6 +71,38 @@ def requested_authn_context
       end
     end
 
+    def requested_authn_contexts
+      if authn_request? && authn_context_nodes.length > 0
+        authn_context_nodes.map(&:content)
+      else
+        nil
+      end
+    end
+
+    def requested_ial_authn_context
+      return nil if requested_authn_contexts.nil?
+
+      ial_prefix = 'http://idmanagement.gov/ns/assurance/ial'.freeze
```

Comment:
these just get turned into regexes...what if we made constants as regexes? And anchored them?

Combined with the previous comment of always returning an array:

```ruby
IAL_PREFIX = %r{^http://idmanagement.gov/ns/assurance/ial}.freeze
LOA_PREFIX = %r{^http://idmanagement.gov/ns/assurance/loa}.freeze

def requested_ial_authn_context
  requested_authn_contexts.first do |classref|
    IAL_PREFIX.match?(classref) || LOA_PREFIX.match?(classref)
  end
end
```

Reply:
cool. had to do:
```
  requested_authn_contexts.select do |classref|
    IAL_PREFIX.match?(classref) || LOA_PREFIX.match?(classref)
  end.first
```
but much cleaner!